### DISCUSSION TITLE
Treatment of spinor bilinears in FeynCalc

### DISCUSSION THREAD

**Discussion by vsht** [FeynCalc/feyncalc#60] [Q&A] [answered]

**Comment by mescia** [comment]
I am using FeynCalc 9.3.1 (stable version). 

From DiracSimplify[Spinor[Momentum[ps], ms].Spinor[Momentum[ps], ms]] I get ms, no spinor anymore.   Why do the spinors disappear? 

Thanks, Federico.

**Reply by vsht** [reply]
Because it corresponds to `ubar(p,m) u(p,m)`, which is `2m` in the relativistic normalization.

**Reply by vsht** [reply]
Perhaps I misunderstood the question. Was it about whether the spinors can be kept despite being equal to `2m`?

**Comment by mescia** [comment, marked as answer]
I did not formulate my question well.  Sorry for this.

I am doing an operator matching in the collinear regions.
My idea was to keep all the spinor structures for 4 and 6 fermions. But at
some point I noticed I had no spinors in my amplitudes (panic!). This was
because as you correctly noticed Feyncalc automatically evaluates the Dirac
structure of tescalar densities, ub(p).ub(p) as 2m.  This is only done for
scalar dentisities.
Unfortunately in the matching I don't need to directly evaluate amplitudes
because quarks are not free. I need only to collect all the billinears and
then I will calculate the corresponding matrix elements using form factors.
Technically, to keep the spinor structures up to then makes the calculation
symmetric with respect to all the other fermion bilinears. and it is usell
to build the operator basis in a large calcualtion.

One more thing, I used DiracSimplify to apply the equation of motion after
divergences in the loop are cancelled out. Then, DiracSimply apply the
equation of motion reducing my operator basis. But at this point it also
evaluates the scalar density and spinor disappears which is complictaed to
classify the operator basis. Is there a Flag in DiracSimply to stop it?

thanks very much and sorry for the confusion. Federico.







On Mon, 7 Sep 2020 at 10:27, Vladyslav Shtabovenko <notifications@github.com>
wrote:

> Perhaps I misunderstood the question. Was it about whether the spinors can
> be kept despite being equal to 2m?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/FeynCalc/feyncalc/discussions/60#discussioncomment-63480>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEBUE74WKJRS4X3ASC46UELSESKNBANCNFSM4PLYUA3Q>
> .
>
>
> Aquest missatge, i els fitxers adjunts que hi pugui haver, pot contenir
> informació confidencial o protegida legalment i s’adreça exclusivament a la
> persona o entitat destinatària. Si no consteu com a destinatari final o no
> teniu l’encàrrec de rebre’l, no esteu autoritzat a llegir-lo, retenir-lo,
> modificar-lo, distribuir-lo, copiar-lo ni a revelar-ne el contingut. Si
> l’heu rebut per error, informeu-ne el remitent i elimineu del sistema tant
> el missatge com els fitxers adjunts que hi pugui haver.
>
> Este mensaje, y los ficheros adjuntos que pueda incluir, puede contener
> información confidencial o legalmente protegida y está exclusivamente
> dirigido a la persona o entidad destinataria. Si usted no consta como
> destinatario final ni es la persona encargada de recibirlo, no está
> autorizado a leerlo, retenerlo, modificarlo, distribuirlo o copiarlo, ni a
> revelar su contenido. Si lo ha recibido por error, informe de ello al
> remitente y elimine del sistema tanto el mensaje como los ficheros adjuntos
> que pueda contener.
>
> This email message and any attachments it carries may contain confidential
> or legally protected material and are intended solely for the individual or
> organization to whom they are addressed. If you are not the intended
> recipient of this message or the person responsible for processing it, then
> you are not authorized to read, save, modify, send, copy or disclose any
> part of it. If you have received the message by mistake, please inform the
> sender of this and eliminate the message and any attachments it carries
> from your account.
>
>

-- 



*Federico Mescia*
Catedràtic de Física Teòrica | Universitat de Barcelona
Dept Física Quàntica i Astrofísica & Institut de Ciències del Cosmos

p: +34 9340 21161 o: 6.07, 6ª planta
a: Diagonal 647, Facultat de Física, Barcelona
w: icc.ub.edu e: mescia@ub.edu s: federico.mescia

**Reply by vsht** [reply]
No problem. I see that the question is not a trivial one, which is good.

At the moment it is not really possible to prevent DiracSimplify from doing
this simplification, since it is hard-coded [here](https://github.com/FeynCalc/feyncalc/blob/2688f08f706e8b3ed7a050efbd5c9d7cafeb8b61/FeynCalc/Dirac/DiracSimplify.m#L476).

It is also the only place in the whole package where this replacement is applied.
If it is urgent, you could just comment the code between the lines 476 and 486.

In any case I agree that this is something that should be configurable, especially
since I also would like different normalization conventions to be available via an
option.

It's also not really much work, just a bit of code and a bunch of compulsory unit tests.
Then I would also be able to merge #64 to finally have the Gordon identities built-in.

Let me see if I manage to add this to the dev version within this week.

**Reply by vsht** [reply]
Ok, all done. The new option `SpinorChainEvaluate -> False` in the dev version should do the trick

```
DiracSimplify[
 x SpinorUBarD[p].GAD[mu].SpinorUD[p] + y SpinorUBarD[p].SpinorUD[p], 
 SpinorChainEvaluate -> False, FCE -> True]
```

```
TID[(x SpinorUBarD[p].GAD[nu].SpinorUD[p] + 
    y SpinorUBarD[p].SpinorUD[p]) FVD[l, mu] FAD[l, l - p], l, 
 SpinorChainEvaluate -> False, FCE -> True]
```

**Comment by mescia** [comment]
Thanks very much. Federico
